### PR TITLE
docs(tree): add cdk tree examples to material-examples

### PR DESCRIPTION
**Pull request by tinayuangao**

### CODE REVIEW

File: src/material-examples/tree-flat/tree-flat-example.css [hunk 1]
```diff
@@ -0,0 +1,6 @@
+.example-tree ul,
+.example-tree li {
+  -webkit-margin-before: 0;
+  -webkit-margin-after: 0;
```
Comment (jelbourn) [inline src/material-examples/tree-flat/tree-flat-example.css]: Why are these prefixed styles necessary?

File: src/material-examples/tree-flat/tree-flat-example.html [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<cdk-tree [dataSource]="dataSource" [treeControl]="treeControl">
+  <cdk-tree-node *cdkTreeNodeDef="let node" cdkTreeNodePadding class="demo-tree-node">
+    <a [attr.aria-label]="'toggle ' + node.filename" cdkTreeNodeToggle class="demo-tree-toggle">
```
Comment (jelbourn) [inline src/material-examples/tree-flat/tree-flat-example.html]: Anchor elements should only be used for navigation (should always have an `href`). For actions, you should use a styled `<button>`.

File: src/material-examples/tree-flat/tree-flat-example.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+    // Notify the change.
+    this.dataChange.next(data);
+  }
+
+  /**
+   * Build the file structure tree. The `value` is the Json object, or a sub-tree of a Json object.
+   * The return value is the list of `FileNode`.
+   */
+  buildFileTree(value: any, level: number) {
```
Comment (jelbourn) [inline src/material-examples/tree-flat/tree-flat-example.ts]: Can the return type on this be more specific?

**Comment by tinayuangao** [comment]
@jelbourn Addressed comments and also updated the material examples.

**Pull request merged**